src/macros/arithmetics.rs

```diff
@@ -16,19 +16,22 @@
 /// - `trait_function`: the function the trait implements
 /// (e.g. add for [`Add`], ...).
 /// - `type`: the type the trait is implemented for (e.g. [`Z`], [`Q`])
+/// - `other_type`: the type the second part of the computation.
+/// - `output_type`: the type of the result.
 ///
 /// Returns the owned Implementation code for the [`*trait*`]
 /// trait with the signature:
 ///
-/// ```impl *trait* for *type*```
+/// ```impl *trait<*other_type*>* for *type*```
 macro_rules! arithmetic_trait_borrowed_to_owned {
-    ($trait:ident, $trait_function:ident, $type:ident) => {
-        impl $trait for $type {
-            type Output = $type;
+    ($trait:ident, $trait_function:ident, $type:ident, $other_type:ident, $output_type:ident) => {
+        #[doc(hidden)]
+        impl $trait<$other_type> for $type {
+            type Output = $output_type;
 
             paste::paste! {
-                #[doc = "Documentation at [`" $type "::" $trait_function "`]."]
-                fn $trait_function(self, other: Self) -> Self::Output {
+                #[doc = "Documentation at [`" $output_type "::" $trait_function "`]."]
```

Comment:
Shouldn't this be $type?
```suggestion
                #[doc = "Documentation at [`" $type "::" $trait_function "`]."]
```

Reply:
no, the documentation should be shown at the output type (the suggestion would link Z+Q to Z and Q+Z to Q, which is unintuitive), maybe an update of the documentation with new examples is needed

Reply:
should be a todo for Q and Zq